lib/metadataTypes/TriggeredSend.js

```diff
@@ -127,17 +127,16 @@ class TriggeredSend extends MetadataType {
         // remove IsPlatformObject, always has to be 'false'
         delete metadata.IsPlatformObject;
         // folder
-        try {
-            this.setFolderPath(metadata);
-        } catch {
-            Util.logger.verbose(
-                ` ☇ skipping ${this.definition.typeName} '${metadata.Name}'/'${metadata.CustomerKey}': Could not find folder.`
-            );
-            // do not save this TSD because it would not be visible in the user interface
-            return;
-        }
-
-        // email
+        // try {
+        //     this.setFolderPath(metadata);
+        // } catch {
+        //     Util.logger.verbose(
+        //         ` ☇ skipping ${this.definition.typeName} '${metadata.Name}'/'${metadata.CustomerKey}': Could not find folder.`
+        //     );
+        //     // do not save this TSD because it would not be visible in the user interface
+        //     return;
+        // }
```

Comment:
@JoernBerkefeld returned metadata is undefined and refresh does not work if we have this piece of code executed. Do we need this for refresh? I don't understand why do we need to set folder if this is not a retrieve?

Reply:
Hm, I'm having a hard time checking this on my phone but: this method is only executed during a retrieve. Not during caching.

Therefore, either our refresh logic retrieves (instead of caches) for no reason or we needed the download including saving to disk (aka retrieve) for the refresh somehow.

Reply:
We execute it here to validate the TS:

https://github.com/Accenture/sfmc-devtools/blob/4093f49bfe64f026eaee372dce83ac5f02768789/lib/metadataTypes/TriggeredSend.js#L277

Reply:
And here we should check if type==="folder" or cache...

Otherwise the necessary folder subtypes for TS are not cached

Reply:
@JoernBerkefeld somehow an error was thrown in searchForField() function that was not caught by try catch on lines 103-119. I've added a try catch and that helped to avoid the problem, refresh completes successfully. 
I don't entirely understand what kind of folders are we talking about when it comes to TriggeredSends, maybe these ones (the corresponding triggered send is located in the Journey sends -> journey)? Is there a problem somewhere with caching that those folders aren't cached?
![image](https://github.com/Accenture/sfmc-devtools/assets/79999614/027f66d7-f0d6-4fd0-b94d-686456759a53)
This is what it looks like when an outer try catch has been added:
![image](https://github.com/Accenture/sfmc-devtools/assets/79999614/c81dc122-aad0-423e-82e2-f616ffa6bc27)

Reply:
pushed it forward run without errors
![image](https://github.com/Accenture/sfmc-devtools/assets/1917227/865cbe86-5d65-41c8-931a-15abf9a8faa9)